include/openthread/thread.h

```diff
@@ -592,6 +592,21 @@ const char *otThreadGetNetworkName(otInstance *aInstance);
  */
 otError otThreadSetNetworkName(otInstance *aInstance, const char *aNetworkName);
 
+/**
+ * Set the Thread Network Name. [DEPRECATED]
+ *
+ * This function is identical to otThreadSetNetworkName() except that it permits setting a zero
+ * length network name. This should not be used for new applications/systems.
+ *
+ * @param[in]  aInstance     A pointer to an OpenThread instance.
+ * @param[in]  aNetworkName  A pointer to the Thread Network Name.
+ *
+ * @retval OT_ERROR_NONE           Successfully set the Thread Network Name.
+ * @retval OT_ERROR_INVALID_STATE  Thread protocols are enabled.
+ *
+ */
+otError otThreadSetDeprecatedNetworkName(otInstance *aInstance, const char *aNetworkName);
```

Comment:
Rather than creating a new API, how about we create a build option that disables the zero-length check? Something like `OPENTHREAD_CONFIG_NETWORK_NAME_ZERO_LENGTH_ENABLE`, which is set to false by default?